data/src/main/java/io/gnosis/data/repositories/TransactionRepository.kt

```diff
@@ -33,18 +32,47 @@ class TransactionRepository(
     suspend fun submitConfirmation(safeTxHash: String, signedSafeTxHash: String): TransactionDetails =
         gatewayApi.submitConfirmation(safeTxHash, TransactionConfirmationRequest(signedSafeTxHash))
 
+    suspend fun proposeTransaction(
+        safeAddress: Solidity.Address,
+        value: BigInteger = BigInteger.ZERO,
+        data: String? = null,
+        nonce: BigInteger,
+        operation: Operation = Operation.CALL,
+        safeTxGas: Long = 0,
+        baseGas: Long = 0,
+        gasPrice: Long = 0,
+        gasToken: Solidity.Address = "0x00".asEthereumAddress()!!,
```

Comment:
I would use the more standard `0x0000000000000000000000000000000000000000`, as the web interface allows `0x0` which is already unconventional.

Reply:
why does it matter here? Type is Solidity.Address so it will be converted to the same value. I would prefer shorter notation here.